src/components/fields/input-field.vue

```diff
@@ -13,20 +13,30 @@
         </template>
       </template>
 
-      <input
-        v-bind="computedAttrs"
-        :type="type"
-        :value="modelValue"
-        :style="inputFieldStyles"
-        :disabled="disabled"
-        :tabindex="tabindex"
-        :min="minValue"
-        class="input-field__input"
-        autocomplete="off"
-        autocorrect="off"
-        autocapitalize="off"
-        spellcheck="false"
-      >
+      <div class="input-field__wrapper">
+        <div class="leading-icon">
+          <span class="iconLeading"></span>
+        </div>
+
+        <input
+          v-bind="computedAttrs"
+          :type="type"
+          :value="modelValue"
+          :style="inputFieldStyles"
+          :disabled="disabled"
+          :tabindex="tabindex"
+          :min="minValue"
+          class="input-field__input"
+          autocomplete="off"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+        >
+
+        <div v-if="currencyCode" class="trailing-icon">
+          <span class="iconTrailing">{{ currencyCode }}</span>
+        </div>
```

Comment:
Same about slots. And `InputField` component should know nothing about anything like `currencyCode`. It should only know that something was passed using a slot